### PR TITLE
Only support Python >= 3.5

### PR DESCRIPTION
**Pull request by LSchueler**
As discussed in issue #40, I've now added the hard Python version requirements for the currently supported versions to the setup.py
Next commits, we can change the line to
`python_requires=">=3.5.*",`

### CODE REVIEW

**Comment by LSchueler** [comment]
As the Window testing on appveyor will be replaced by TravisCI, I won't bother with the failed appveyor builds.

File: setup.py [hunk 1]
```diff
@@ -40,19 +40,6 @@ def find_version(*file_paths):
...
-class _TemporaryDirectory:
-    def __enter__(self):
-        self.dir_name = tempfile.mkdtemp()
-        return self.dir_name
-
-    def __exit__(self, exc_type, exc_value, traceback):
-        shutil.rmtree(self.dir_name)
-
-
-TemporaryDirectory = getattr(
```
Comment (MuellerSeb) [inline setup.py]: You have to import TemporaryDirectory from tempfile now or use it by ``tempfile.TemporaryDirectory``

Comment (LSchueler) [reply setup.py]: ups, thanks

**Comment by MuellerSeb** [comment]
The diff for ``.travis.yml`` makes no sense. They are identical on both branches ?!?!

**Pull request merged**